### DISCUSSION TITLE
Cannot initialize project using MySQL on Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by BeckySalvage** [ROBelgium/MSNoise#254] [Q&A] [answered]
Hi,
I followed the install instructions on Ubuntu 20.04 and successfully connected to MySQL. However, when I run msnoise db init --help, I get the following error: 

msnoise db init --help
Usage: msnoise [OPTIONS] COMMAND [ARGS]...
Try "msnoise --help" for help.

Error: No such command "db"

msnoise is correctly installed as msnoise --help brings up the help menu. Has anyone else had this issue? Since I will need to process a number of years worth of data, I think I need to use MySQL rather than the lite version. 

Any help would be really appreciated!
Thanks,
Becky

**Comment by ThomasLecocq** [comment, marked as answer]
Hi Becky,

How did you install msnoise ? What is the output of ``msnoise --help``

**Reply by BeckySalvage** [reply]
Hey Tom,

I installed via the terminal so:

conda install -c conda-forge flask-admin flask-wtf markdown folium pymysql logbook
conda install -c conda-forge obspy
sudo apt-get install mysql-server
Then created user in mysql-workbench
pip install msnoise

The output of msnoise --help is:

msnoise --help
Usage: msnoise [OPTIONS] COMMAND [ARGS]...

Options:
  -t, --threads INTEGER  Number of threads to use (only affects modules that
                         are designed to do parallel processing)
  -d, --delay INTEGER    In the case of multi-threading, defines the number of
                         seconds to wait before lauching the next thread.
                         Defaults to [1] second
  -c, --custom           Use custom  file for plots. To use this, copy the
                         plot script here and edit it.
  -v, --verbose
  --help                 Show this message and exit.

Commands:
  admin
  bugreport           This command launches the Bug Report script.
  compute-cc          Computes the CC jobs (based on the "New Jobs"...
  compute-dtt         Computes the dt/t jobs based on the new MWCS data
  compute-mwcs        Computes the MWCS based on the new stacked data
  compute-stretching  [experimental] Computes the stretching based on the...
  config              This command should now only be used to use the...
  info                Outputs general information about the current install...
  install             This command launches the installer.
  ipython             Launches an ipython notebook in the current folder
  new-jobs            Determines if new CC jobs are to be defined
  p                   Short cut for plugins
  plot                Top level command to trigger different plots
  plugin              Runs a command in a named plugin
  populate            Rapidly scan the archive filenames and find...
  reset               Resets the job to "T"odo.
  scan-archive        Scan the archive and insert into the Data
                      Availability...
  stack               Stacks the [REF] and/or [MOV] windows
  test                Runs the test suite, should be executed in an empty...
  upgrade-db          Upgrade the database from previous to a new version.

Not sure if it is of any use but the version of MySQL is 8.0.28-1ubuntu20.04.

Thanks for your help!

Becky

**Reply by ThomasLecocq** [reply]
this is wild, could you try :

pip uninstall msnoise
pip install http://msnoise.org/master.zip

this should install the latest version. It looks like you're running a verrrryyyy old version of msnoise (not 1.6.2 from pip), the error is not linked to mysql.

**Reply by BeckySalvage** [reply]
Oh weird - when it uninstalled it was msnoise 1.5 so somewhere it obviously got confused.

Uninstalling and installing as you said above installs: 

```
Successfully built msnoise
Installing collected packages: msnoise
Successfully installed msnoise-0.0.0+archive
```
Should I be worried that it is not v1.6.2? 

I then tried `msnoise db init --help` which works and prints out the help file :) 

However, if I try `msnoise db init --tech 2` I get an error: 
`
sqlalchemy.exc.OperationalError: (pymysql.err.OperationalError) (2003, "Can't connect to MySQL server on 'None' ([Errno -2] Name or service not known)")`

Should I be somehow forcing msnoise to connect to the SQL database just created (called msnoise) or to the schema created (msnoise1)? 

As always, thanks for all your help!

**Reply by ThomasLecocq** [reply]
Good it's solved. This version has more interesting features than 1.6.2 (almost 2y of dev ahead), so if you wanna try it, it's better :-)

About the db init command, try not passing --tech =2 and answer the questions, it should be OK (seems "host" defaults to "None" which is not OK.

**Reply by BeckySalvage** [reply]
That worked, thanks so much!! :)
